pandas/_libs/algos_common_helper.pxi.in

```diff
@@ -65,7 +65,8 @@ def ensure_{{name}}(object arr, copy=True):
         if (<ndarray>arr).descr.type_num == NPY_{{c_type}}:
             return arr
         else:
-            return arr.astype(np.{{dtype}}, copy=copy)
+            # equiv: arr.astype(np.{{dtype}}, copy=copy)
+            return cnp.PyArray_Cast(<ndarray>arr, cnp.NPY_{{c_type}})
```

Comment:
If I'm interpreting this correctly now it always copies? https://numpy.org/devdocs/reference/c-api/array.html#c.PyArray_Cast

If that's correct can `copy=True` be removed in the signature?

Reply:
We only get here when the dtypes don't match, so i think a copy needs to be made regardless of the copy keyword.

Reply:
Got it, so can the `copy=True` in the signature be removed then as well?

Reply:
i want to investigate whether the fact that we _dont_ make a copy in the matching-dtype case a few lines up causes a subtle bug anywhere.  If not, then sure.

Reply:
Were you thinking of investigating in this PR or a follow up?

Reply:
was planning to do that separately